test/e2e/service.go

```diff
@@ -2350,3 +2371,33 @@ func execSourceipTest(f *framework.Framework, c *client.Client, ns, nodeName, se
 
 	return execPodIp, sourceIp
 }
+
+func getLoadBalancerName(service *api.Service) string {
+	//GCE requires that the name of a load balancer starts with a lower case letter.
+	ret := "a" + string(service.UID)
+	ret = strings.Replace(ret, "-", "", -1)
+	//AWS requires that the name of a load balancer is shorter than 32 bytes.
+	if len(ret) > 32 {
+		ret = ret[:32]
+	}
+	return ret
+}
+
+func cleanupServiceGCEResources(loadBalancerName string) {
+	if pollErr := wait.Poll(5*time.Second, lbCleanupTimeout, func() (bool, error) {
+		if err := framework.CleanupGCEResources(loadBalancerName); err != nil {
```

Comment:
the behavior you mentioned existing in current code.
if you look at https://github.com/kubernetes/kubernetes/blob/master/test/e2e/service.go#L748-L766, the last step of the it is changing the service type back to cluster ip and waiting for the LB deletion, so the leak is only possible when error occurs during the service LB type being changed, or some panic happened before that. (we might need wrap this block and put to defer).
In addition, we need to clean the related resource in case the above description does not behave correctly.
There are two service being set LB type for the whole service e2e, tcp service, udp service for `mutability-test` , and `external-local` service, and is not reused in different It(). Only these three service LB names are put to the string array, so it results same in AfterEach or in last resort.